common/tools/dev-tool/src/commands/run/extract-api.ts

```diff
@@ -0,0 +1,150 @@
+    return false;
+  }
+
+  log.debug(`  mainEntryPointFilePath: ${extractorConfigObject.mainEntryPointFilePath}`);
+  log.debug(`  publicTrimmedFilePath: ${extractorConfigObject?.dtsRollup?.publicTrimmedFilePath}`);
+  log.debug(`  apiJsonFilePath: ${extractorConfigObject.docModel?.apiJsonFilePath}`);
+  log.debug(`  reportFileName: ${extractorConfigObject.apiReport?.reportFileName}`);
+  log.debug(`  reportTempFolder: ${extractorConfigObject.apiReport?.reportTempFolder}`);
+
+  let succeed = true;
+  // sub path exports extraction
+  const exports = packageJson["exports"];
+  if (exports) {
+    for (const exprt of Object.keys(exports)) {
+      log.debug(`extracting Api for "${exprt}"`);
+      const suffix = getSuffix(exprt);
+      const newPublicTrimmedPath = extractorConfigObject.dtsRollup.publicTrimmedFilePath.replace(
+        ".d.ts",
+        `-${suffix}.d.ts`
+      );
+      const newApiJsonPath = extractorConfigObject.docModel?.apiJsonFilePath?.replace(
+        ".api.json",
+        `-${suffix}.api.json`
+      );
+      const newApiReportName = extractorConfigObject.apiReport?.reportFileName?.replace(
+        ".api.md",
+        `-${suffix}.api.md`
+      );
+      const newDtsRollupOptions: IConfigDtsRollup = {
+        ...extractorConfigObject.dtsRollup,
+        publicTrimmedFilePath: newPublicTrimmedPath,
+      };
+      const newDocModel: IConfigDocModel = {
+        ...extractorConfigObject.docModel,
+        enabled: true,
+        apiJsonFilePath: newApiJsonPath,
+      };
+      const newApiReport: IConfigApiReport = {
+        ...extractorConfigObject.apiReport,
+        enabled: true, // or should we disable? not sure if we need look at multiple api.md
+        reportFileName: newApiReportName,
+      };
+      const updatedConfigObject: IConfigFile = {
+        ...extractorConfigObject,
+        dtsRollup: newDtsRollupOptions,
+        docModel: newDocModel,
+        apiReport: newApiReport,
+        mainEntryPointFilePath: exports[exprt].types,
+      };
+      const extractorConfig = ExtractorConfig.prepare({
+        configObject: updatedConfigObject,
+        configObjectFullPath: apiExtractorJsonPath,
+        packageJsonFullPath: packageJsonPath,
+      });
+      const extractorResult: ExtractorResult = Extractor.invoke(extractorConfig, {
+        // Equivalent to the "--local" command-line parameter
+        localBuild: true,
+        // Equivalent to the "--verbose" command-line parameter
+        showVerboseMessages: true,
+      });
+      if (extractorResult.succeeded) {
+        log.debug(`API Extractor completed successfully for ${exprt} export`);
+      } else {
+        succeed = false;
+        log.error(
+          `API Extractor completed with ${extractorResult.errorCount} errors` +
+            ` and ${extractorResult.warningCount} warnings`
+        );
+      }
+    }
+  } else {
+    // normal extraction
+    const config = ExtractorConfig.prepare({
```

Comment:
Is there an opportunity to abstract some of this logic between both the exports case and the non-exports case?

Reply:
good point! I should've done that earlier.